crates/node/src/lib.rs

```diff
@@ -301,15 +305,31 @@ impl Node {
                 author_id,
                 root_hash,
                 artifact,
+                sending_identity,
             } => {
-                self.handle_state_delta(
-                    source,
-                    context_id,
-                    author_id,
-                    root_hash,
-                    artifact.into_owned(),
-                )
-                .await?;
+                let identities = self.ctx_manager.get_context_owned_identities(context_id)?;
+
+                let Some(our_identity) = identities.into_iter().choose(&mut thread_rng()) else {
+                    bail!("no identities found for context: {}", context_id);
+                };
+
+                let our_sending_key = self
+                    .ctx_manager
+                    .get_own_signing_key(&context_id, &our_identity)?;
+
+                let shared_key = SharedKey::new(
+                    &our_sending_key,
+                    &VerifyingKey::from_bytes(&sending_identity)?,
+                );
```

Comment:
when receiving a state delta broadcast, we'll be using the sender key as opposed to deriving a shared key